ambry-cloud/src/main/java/com.github.ambry.cloud/AzureFindToken.java

```diff
@@ -0,0 +1,193 @@
+    version = DEFAULT_VERSION;
+  }
+
+  /**
+   * Create {@link AzureFindToken} from provided values.
+   * @param startContinuationToken
+   * @param endContinuationToken
+   * @param index
+   * @param totalItems
+   * @param azureTokenRequestId
+   */
+  public AzureFindToken(String startContinuationToken, String endContinuationToken, int index, int totalItems,
+      String azureTokenRequestId) {
+    this.startContinuationToken = startContinuationToken;
+    this.endContinuationToken = endContinuationToken;
+    this.index = index;
+    this.totalItems = totalItems;
+    this.azureTokenRequestId = azureTokenRequestId;
+    this.version = DEFAULT_VERSION;
+  }
+
+  /**
+   * Constructor to create a {@link AzureFindToken} with specified token values and specified version.
+   * @param startContinuationToken
+   * @param endContinuationToken
+   * @param index
+   * @param totalItems
+   * @param azureTokenRequestId
+   * @param version
+   */
+  public AzureFindToken(String startContinuationToken, String endContinuationToken, int index, int totalItems,
+      String azureTokenRequestId, short version) {
+    this.startContinuationToken = startContinuationToken;
+    this.endContinuationToken = endContinuationToken;
+    this.index = index;
+    this.totalItems = totalItems;
+    this.azureTokenRequestId = azureTokenRequestId;
+    this.version = version;
+  }
+
+  /**
+   * Deserialize {@link AzureFindToken} object from input stream.
+   * @param inputStream {@link DataOutputStream} to deserialize from.
+   * @return {@link AzureFindToken} object.
+   * @throws IOException
+   */
+  public static AzureFindToken fromBytes(DataInputStream inputStream) throws IOException {
+    short version = inputStream.readShort();
+    String startContinuationToken = extractStringFromStream(inputStream);
+    String endContinuationToken = extractStringFromStream(inputStream);
+    int index = inputStream.readInt();
+    int totalItems = inputStream.readInt();
+    String azureTokenRequestId = extractStringFromStream(inputStream);
+    return new AzureFindToken(startContinuationToken, endContinuationToken, index, totalItems, azureTokenRequestId,
+        version);
+  }
+
+  /**
+   * Extract string from the {@link DataInputStream}.
+   * @param inputStream {@link DataInputStream} to extract String from.
+   * @return extracted String from {@code inputStream}.
+   * @throws IOException
+   */
+  private static String extractStringFromStream(DataInputStream inputStream) throws IOException {
+    int size = inputStream.readInt();
+    byte[] bytes = new byte[size];
+    inputStream.read(bytes);
+    return new String(bytes);
+  }
+
+  public byte[] toBytes() {
+    byte[] buf = new byte[size()];
+    ByteBuffer bufWrap = ByteBuffer.wrap(buf);
+    bufWrap.putShort(version);
+    bufWrap.putInt(startContinuationToken.length());
+    bufWrap.put(startContinuationToken.getBytes());
+    bufWrap.putInt(endContinuationToken.length());
```

Comment:
Use `Utils.serializeString` for all of the size prefixed string type things here. It handles edge cases around character encodings.

Reply:
done.